ci-operator/step-registry/cucushift/installer/check/ibmcloud/custom-endpoints/cucushift-installer-check-ibmcloud-custom-endpoints-commands.sh

```diff
@@ -28,11 +15,100 @@ function get_service_endpoint() {
     echo "${service_endpoint}"
 }
 
+function exists_in()
+{
+	local val=$1
+	shift
+	local val_list=("$@")
+	local match=1
+	for v in "${val_list[@]}";
+	do
+	    if [[ "$val" = "$v" ]]; then
+	        match=0
+	        break
+	    fi
+	done
+	return $match
+}
+
+function isPreVersion() {
+  local REQUIRED_OCP_VERSION="$1"
+  local isPre
+  version=$(oc version -o json | jq -r '.openshiftVersion' | cut -d '.' -f1,2)
+  echo "get ocp version: ${version}"
+
+  isPre=0
+  if [ -n "${version}" ] && [ "$(printf '%s\n' "${REQUIRED_OCP_VERSION}" "${version}" | sort --version-sort | head -n1)" = "${REQUIRED_OCP_VERSION}" ]; then
+    isPre=1
+  fi
+  return $isPre
+}
+
+function check_ep_names() {
+    local isPre418=$1
+    local output expString
+    output=$(openshift-install explain installconfig.platform.ibmcloud.serviceEndpoints)
+    if isPre418; then
```

Comment:
Actually you do not need some complicated logic to handle version difference, you just to prepare target service  list array for different versions, just like https://github.com/openshift/release/pull/58984/files#r1859767876, then load the array to assemble the expected string.